tests/integration/HACKING.md

```diff
@@ -0,0 +1,468 @@
+	.param('file_id', 'File Id to run "some_task" on', dataType='string')))
+def some_rest_endpoint(file_id, **params):
+	# Create a token with permissions of the current user.
+    token = ModelImporter.model('token').createToken(
+        user=self.getCurrentUser())
+
+    # Launch the asynchronous task
+	async_result = some_task.delay(file_id, girder_client_token=token)
+	
+	return async_result.job
+```
+
+Consumer side task:
+```python
+from girder_worker.app import app
+
+@app.task(bind=True)
+def some_task(self, file_id):
+	# Note: girder_client is available as an attribute of 'self'
+	# self (the task object) will be passed to the function if
+	# bind is set to True in the task decorator.
+	self.girder_client.downloadFile(file_id, '/tmp/file')
+	with open('/tmp/file', 'rb') as f:
+	    # Do something with contents of f
+
+```
+
+In this example the producer side REST endpoint creates a token with the scope of the current user, then passes the REST argument ```file_id``` and the token to the task object's ```.delay(...)``` function.  On the consumer side,  the task has access to the GirderClient instance through ```self.girder_client```. To get access to ```self``` the task must be decorated with ```@app.task(bind=True)``` (See [bound tasks](http://docs.celeryproject.org/en/latest/userguide/tasks.html#bound-tasks) for more information). 
+
+# How do I write a custom Celery task for Girder Worker?
+
+*Note that this is taken almost directly from [Writing your own plugin](http://girder-worker.readthedocs.io/en/latest/developer-docs.html#writing-your-own-plugin)*
+
+Adding additional tasks to the girder\_worker infrastructure is easy and takes three steps. 
+
+1. Creating tasks
+2. Creating a plugin class 
+3. Adding a girder\_worker\_plugins [entry point](http://setuptools.readthedocs.io/en/latest/setuptools.html#dynamic-discovery-of-services-and-plugins) to your setup.py.
+
+Creating tasks follows the [standard celery conventions](http://docs.celeryproject.org/en/latest/userguide/tasks.html#basics). The only difference is the celery application that decorates the function should be imported from girder_worker.app. E.g.:
+
+```python
+from girder_worker.app import app
+
+@app.task
+def fibonacci(n):
+    if n == 1 or n == 2:
+        return 1
+    return fibonacci(n-1) + fibonacci(n-2)
+	
+```
+
+Each plugin must define a plugin class the inherits from ```girder_worker.GirderWorkerPluginABC```. GirderWorkerPluginABC's interface is simple. The class must define an \_\_init\_\_ function and a task\_imports function. \_\_init\_\_ takes the girder\_worker's celery application as its first argument. This allows the plugin to store a reference to the application, or change configurations of the application as necessary. The task\_imports function takes no arguments and must return a list of the package paths (e.g. importable strings) that contain the plugin’s tasks. As an example:
```

Comment:
is `GirderWorkerPluginABC` a placeholder or a real name?  if a real name, it needs some explanation.